rust/tvm-rt/src/function.rs

```diff
@@ -196,8 +204,8 @@ impl TryFrom<RetValue> for Function {
     }
 }
 
-impl<'a> From<Function> for ArgValue<'a> {
-    fn from(func: Function) -> ArgValue<'a> {
+impl<'a> From<&Function> for ArgValue<'a> {
```

Comment:
Function implements drop which looks like it frees the function? I think this needs a similar treatment as the other `From` impls.

Reply:
Looks good. Thought maybe ToFunction or Function's handles might be able to produce a dangling reference and I don't think it can.

Reply:
I think its probably still good to constrain here, the problem is the 'a is sort of a phantom lifetime and we should probably use the `impl` to tie them together.